### ISSUE TITLE
[Bug]: Unexpected token o in JSON at position 1

### ISSUE THREAD

**Issue by AvaYnE2** [pingdotgg/uploadthing#362]
### What happened?

I'm encountering a consistent issue in both my main Next.js project and an example project from UploadThing named "minimal-pagedir." During the file upload process, I receive the following error message:
```
- error Error [SyntaxError]: Unexpected token o in JSON at position 1
```

### What did you expect to happen?

I expected the files to upload successfully without encountering JSON parsing errors.

### Version

```
"dependencies": {
    "@uploadthing/react": "5.6.0",
    "next": "13.4.4",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "uploadthing": "5.6.0"
  },
  "devDependencies": {
    "@types/node": "18.16.0",
    "@types/react": "18.2.8",
    "@types/react-dom": "18.2.4",
    "typescript": "^5.1.6"
  }
```

### Reproduction Steps

1. Install UploadThing in a Next.js project and configure it according to the official documentation.
2. Try uploading a file in the main project.
3. Separately, clone the UploadThing example project named "minimal-pagedir."
4. Try uploading a file in the example project as well.

### Link to reproduction

https://github.com/pingdotgg/uploadthing/tree/main/examples/minimal-pagedir

### Relevant log output

```bash
- error Error [SyntaxError]: Unexpected token o in JSON at position 1
    at JSON.parse (<anonymous>)
    at IncomingMessage.json (file:///G:/coding/uploadthing/uploadthing/examples/minimal-pagedir/node_modules/uploadthing/dist/next-legacy.mjs:27:16)
    at file:///G:/coding/uploadthing/uploadthing/examples/minimal-pagedir/node_modules/uploadthing/dist/chunk-UHCG7WYR.mjs:371:55
    at file:///G:/coding/uploadthing/uploadthing/examples/minimal-pagedir/node_modules/uploadthing/dist/next-legacy.mjs:24:28
    at G:\coding\uploadthing\uploadthing\examples\minimal-pagedir\node_modules\next\dist\server\api-utils\node.js:463:16
    at G:\coding\uploadthing\uploadthing\examples\minimal-pagedir\node_modules\next\dist\server\lib\trace\tracer.js:117:36
    at NoopContextManager.with (G:\coding\uploadthing\uploadthing\examples\minimal-pagedir\node_modules\next\dist\compiled\@opentelemetry\api\index.js:1:7057)
    at ContextAPI.with (G:\coding\uploadthing\uploadthing\examples\minimal-pagedir\node_modules\next\dist\compiled\@opentelemetry\api\index.js:1:516)
    at NoopTracer.startActiveSpan (G:\coding\uploadthing\uploadthing\examples\minimal-pagedir\node_modules\next\dist\compiled\@opentelemetry\api\index.js:1:18086)
    at ProxyTracer.startActiveSpan (G:\coding\uploadthing\uploadthing\examples\minimal-pagedir\node_modules\next\dist\compiled\@opentelemetry\api\index.js:1:18847)
    at G:\coding\uploadthing\uploadthing\examples\minimal-pagedir\node_modules\next\dist\server\lib\trace\tracer.js:106:107
    at NoopContextManager.with (G:\coding\uploadthing\uploadthing\examples\minimal-pagedir\node_modules\next\dist\compiled\@opentelemetry\api\index.js:1:7057)
    at ContextAPI.with (G:\coding\uploadthing\uploadthing\examples\minimal-pagedir\node_modules\next\dist\compiled\@opentelemetry\api\index.js:1:516)
    at NextTracerImpl.trace (G:\coding\uploadthing\uploadthing\examples\minimal-pagedir\node_modules\next\dist\server\lib\trace\tracer.js:106:32)
    at apiResolver (G:\coding\uploadthing\uploadthing\examples\minimal-pagedir\node_modules\next\dist\server\api-utils\node.js:461:63)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async DevServer.runApi (G:\coding\uploadthing\uploadthing\examples\minimal-pagedir\node_modules\next\dist\server\next-server.js:664:9)
    at async Object.fn (G:\coding\uploadthing\uploadthing\examples\minimal-pagedir\node_modules\next\dist\server\next-server.js:1130:35)
    at async Router.execute (G:\coding\uploadthing\uploadthing\examples\minimal-pagedir\node_modules\next\dist\server\router.js:315:32)
    at async DevServer.runImpl (G:\coding\uploadthing\uploadthing\examples\minimal-pagedir\node_modules\next\dist\server\base-server.js:602:29)
    at async DevServer.run (G:\coding\uploadthing\uploadthing\examples\minimal-pagedir\node_modules\next\dist\server\dev\next-dev-server.js:922:20)
    at async DevServer.handleRequestImpl (G:\coding\uploadthing\uploadthing\examples\minimal-pagedir\node_modules\next\dist\server\base-server.js:533:20) {
  digest: undefined
}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by juliusmarminge** [comment]
Hmm this is strange... I don't see when this would have breaked.

Looks like `req.body` is already JSON parsed:

<img width="767" alt="CleanShot 2023-09-17 at 01 04 58@2x" src="https://github.com/pingdotgg/uploadthing/assets/51714798/921e5dae-accc-4ecc-b095-81fe34c0dda4">

We haven't changed this in forever though so curious why it wasn't broken before?

**Comment by juliusmarminge** [comment]
@AvaYnE2 you can use this canary until the next version is out: `pnpm add uploadthing@5.6.1-canary.baa2a0e`

**Comment by AvaYnE2** [comment]
Hmm weird that it brakes now. Unfortunately, this is the first time I've tried Uploadthing, so I can't really say if there were any related bugs in previous versions. 
But thanks a lot, it works fine with the new version! 
I hope you find the cause.

**Comment by ccutch** [comment]
Hey want to report that I also go this error when i previously didnt in `5.2.x` about 2 months ago

**Comment by juliusmarminge** [comment]
@AvaYnE2 the fix has been released in 5.6.1 now
